bVNC/src/main/java/com/iiordanov/bVNC/RemoteCanvasActivity.java

```diff
@@ -326,15 +327,18 @@ public void onCreate(Bundle icicle) {
             }
         };
 
-        String vvFileName = null;
+        if(Utils.isOpaque(this) || Utils.isRdp(this)) {
+            // TODO: the above if is a hacky solution, find a better one.
+            this.connectionConfigFileName = retrieveConfigFileFromIntent(getIntent());
+        }
         if (Utils.isOpaque(this)) {
-            vvFileName = retrieveVvFileFromIntent(getIntent());
-            connection = getOpaqueConnection(vvFileName);
+            connection = getOpaqueConnection(connectionConfigFileName);
         } else {
             connection = getNonOpaqueConnection();
         }
+        connection.setConnectionConfigFile(this.connectionConfigFileName);
```

Comment:
Please refactor this method into RemoteConnection where it can use member variables rather than all those ternary if/elses.